src/gt4py/cartesian/gtc/debug/debug_codegen.py

```diff
@@ -0,0 +1,354 @@
+        ctx: StencilExtentComputer.Context = StencilExtentComputer().visit(node)
+        return ctx.fields, ctx.blocks
+
+    def initial_declarations(
+        self, stencil: Stencil, field_extents: dict[str, Extent]
+    ) -> dict[str, FieldDecl]:
+        self.body.append("# ===== Domain Description ===== #")
+        self.body.append("i_0, j_0, k_0 = 0,0,0")
+        self.body.append("i_size, j_size, k_size = _domain_")
+        self.body.empty_line()
+        self.body.append("# ===== Temporary Declaration ===== #")
+        symbol_table = self.generate_temp_decls(stencil.declarations, field_extents)
+        self.body.empty_line()
+        self.body.append("# ===== Field Declaration ===== #")
+        symbol_table |= self.generate_field_decls(stencil.params)
+        self.body.empty_line()
+        return symbol_table
+
+    def generate_temp_decls(
+        self, temporary_declarations: list[Temporary], field_extents: dict[str, Extent]
+    ) -> dict[str, FieldDecl]:
+        symbol_table: dict[str, FieldDecl] = {}
+        for declaration in temporary_declarations:
+            self.body.append(self.visit(declaration, field_extents=field_extents))
+            symbol_table[str(declaration.name)] = declaration
+        return symbol_table
+
+    def generate_field_decls(self, declarations: list[Decl]) -> dict[str, FieldDecl]:
+        symbol_table = {}
+        for declaration in declarations:
+            if isinstance(declaration, FieldDecl):
+                self.body.append(
+                    f"{declaration.name} = Field({declaration.name}, _origin_['{declaration.name}'], "
+                    f"({', '.join([str(x) for x in declaration.dimensions])}))"
+                )
+                symbol_table[str(declaration.name)] = declaration
+        return symbol_table
+
+    def generate_run_function(self, stencil: Stencil):
+        function_signature = "def run(*"
+        args = []
+        for param in stencil.params:
+            args.append(self.visit(param))
+        function_signature = ",".join([function_signature, *args])
+        function_signature += ",_domain_, _origin_):"
+        self.body.append(function_signature)
+        self.body.indent()
+
+    def generate_stencil_code(
+        self,
+        stencil: Stencil,
+        block_extents: dict[int, Extent],
+    ):
+        for loop in stencil.vertical_loops:
+            for section in loop.sections:
+                with self.create_k_loop_code(section, loop):
+                    for execution in section.horizontal_executions:
+                        with self.generate_ij_loop(block_extents, execution):
+                            self.visit(execution)
+
+    @contextmanager
+    def create_k_loop_code(self, section: VerticalLoopSection, loop: VerticalLoop) -> Generator:
+        loop_bounds: str = self.visit(section.interval, var="k", direction=loop.loop_order)
+        iterator = "1" if loop.loop_order != LoopOrder.BACKWARD else "-1"
```

Comment:
```suggestion
        increment = "1" if loop.loop_order == LoopOrder.FORWARD else "-1"
```
avoiding negations in conditions generally reduces cognitive load (at least in cases where both, if & else, branches are used)

(name change to be propagated within the function scope)

Reply:
Does not work here, because parallel exists and also needs +1. But we can change the if logic if you want